oauthproxy.go

```diff
@@ -350,6 +351,10 @@ func (p *OAuthProxy) redeemCode(host, code string) (s *sessionsapi.SessionState,
 		s.Email, err = p.provider.GetEmailAddress(s)
 	}
 
+	if s.PreferredUsername == "" {
+		s.PreferredUsername, err = p.provider.GetPreferredUsername(s)
```

Comment:
This error is currently unchecked, and, for most providers, will always error, can we add a check on this error similar to the one below for `s.User`

Reply:
Sure! I've added it in ae34f54c816f1e44985f867c4b271704038ba388.